jobs/haproxy/templates/post-start.erb

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+source /var/vcap/packages/awscli/bosh/runtime.env
+
+set -e
+
+# Register ourselves with the load balancer target groups
+OUR_INSTANCE_ID="$(ec2metadata --instance-id)"
+REGION="ap-southeast-2"
+HTTP_TARGET_GROUP="<%= p('http_target_group') %>"
+HTTPS_TARGET_GROUP="<%= p('https_target_group') %>"
+
+# These should always return 0 exit code, regardless of whether we are already registered
+for TARGET in ${HTTP_TARGET_GROUP} ${HTTPS_TARGET_GROUP} ; do
+    aws elbv2 register-targets --target-group-arn "${TARGET}" --targets "Id=${OUR_INSTANCE_ID}" --region "${REGION}" > /dev/null
+done
+
+# So then wait until we can confirm that we are healthy
+for TARGET in ${HTTP_TARGET_GROUP} ${HTTPS_TARGET_GROUP} ; do
+    while ! aws elbv2 describe-target-health --target-group-arn "${TARGET}" --targets "Id=${OUR_INSTANCE_ID}" --region "${REGION}" | grep '"healthy"' > /dev/null ; do
```

Comment:
Re waiting until we healthy first, no we don't need to, as the load balancer in theory won't mark us as healthy until it says we are, and we then wait for it do do so.

Since these are post-start scripts, that means that all of the start scripts have already finished successfully, so we should be ready to serve traffic immediately upon registration, even when the NLBs misbehave.

Reply:
Ok as long as you think we can trust the NLB to do that much right :) I did notice the gorouter job also has a post-start, but it's just a [wait for the load balancer](https://github.com/cloudfoundry/routing-release/blob/develop/jobs/gorouter/templates/post-start.erb) so should be fine.

Reply:
I wouldn't trust the NLBs as far as I can throw them... but they do seem to have a 5 minute target deregistration delay, and they do seem to take a while to show as healthy.